crates/trie/sparse/src/metrics.rs

```diff
@@ -4,6 +4,7 @@
 use reth_metrics::{metrics::Histogram, Metrics};
 
 /// Metrics for the sparse state trie
+#[cfg(feature = "metrics")]
```

Comment:
let's try to feature gate the entire module in lib.rs, like

```
#[cfg(feature = "metrics")]
mod metrics;
```

Reply:
done :)